src/02-common-principles.md

```diff
@@ -367,6 +367,183 @@ In particular, if a BIDS dataset contains a `derivatives/` sub-directory,
 the contents of that directory may be a heterogeneous mix of BIDS Derivatives
 datasets and non-compliant derivatives.
 
+## BIDS URI: Pointing to files within and outside of BIDS datasets
+
+Throughout the BIDS specification there a meta data fields that contain a
+pointer to a file.
+For example `IntendedFor`, `SpatialReference`, and `AssociatedEmptyRoom`.
+Prior to BIDS version 1.7.0, such pointers were required to be "relative paths".
+However the specification failed to consistently  describe relative to *what*
+these pointers should be specified, and the exact syntax to be used.
+Yet even with perfectly well and consistently defined "relative paths", this
+method of pointing to files becomes impractical when referring to files that
+are outside of a particular file tree (for example, on the Internet), or when
+the location of the data that is being pointed to changes.
+
+For these reasons, starting with BIDS version 1.7.0, using "relative paths" for
+all fields pointing to a separate file is [DEPRECATED][].
+
+Instead, the following [URI][uniform-resource-indicator] scheme SHOULD be
+used to point to files within and outside of BIDS datasets, as elaborated below.
+
+`bids:<dataset-name>:/absolute/path/within/dataset`
+
+Here, `bids:` is the URI scheme, and `<dataset-name>:/absolute/path/within/dataset`
+is the path to the file, consisting of the name of a dataset, a colon, and the absolute
+path within that dataset that MUST start with a forward slash `/`.
+
+The location of a dataset with the name `<dataset-name>` MUST be specified in the
+[`dataset_description.json` file](./03-modality-agnostic-files.md#dataset-description)
+under the `DatasetLinks` field, which is an [object][] of [URIs][], as shown in
+the examples below.
+In brief, `dataset_description.DatasetLinks` contains mappings from strings to URIs
+that point to dataset locations.
+The `<dataset-name>`: `local` is a reserved value that may only be
+used to refer to the root of the current dataset; `local` MUST NOT be a key in the
+`DatasetLinks` object.
```

Comment:
Would it be cleaner to just permit `bids::` to refer to the current dataset, rather than `bids:local:`? Avoids validation of dataset names, pretty visibly different from remote datasets, and could be captured by the same parser as any other BIDS URI by permitting the empty string.

Reply:
I have no objections against this.

> Avoids validation of dataset names

I guess we'd still have to validate that nobody passes an [empty string as a key](https://stackoverflow.com/a/58917082/5201771) in the `DatasetLinks` object then, (because empty string would then be what is now "local")

Reply:
That's true, we would need to make sure all entries are non-empty, which feels like a minimal check. I don't feel very strongly here, but my preference is still empty string over `local`.

Reply:
okay thanks, I'll go with your preference since I am undecided and of the opinion that both will be fine. --> `bids:local:/...` is now going to be `bids::/...`